### PR TITLE
Trade expansion: unlock tree, new stations, QoL/UI stuff

### PR DESCRIPTION
**Pull request by Pink-Chink**
## About The Pull Request

- Reworked the secret inventory unlock feature into favor, which is used for both secret inventories and discovering new stations (without needing to fly to them)
- Buying and selling via the goods menu generates favor equal to half the amount of credits spent/earned
- Offers generate favor equal to the amount of credits earned
- Unlocks are now way easier: 2000 and 4000 are common thresholds compared to 16000+
- Added four new stations: Ningishzida (more Moe stuff), Nauka (tool mods), a second NT station (animals + seeds), and a station for some contractor uplink items
- All existing stations are now discoverable via the favor system or by flying around (8 roundstart + 16 unlockable)
- Tweaked offers and inventories a bit
- Favor and update timer are now shown in the UI
- Made trade bottle subtype to keep new bottles from spawning elsewhere
- Prevents selling items that are worth 0
- Items that have no price can be bought at a base price of 100 (before markup)
- Removed cybernetics trade station for now

Next update: master list of items sold in all beacons, export scanner shows beacon buy and offer prices, more stations

## Why It's Good For The Game

Trade beacon becomes more useful and encourages more crew interaction.

## Changelog
:cl:
add: created favor mechanic to handle unlockables
tweak: unlocks are much easier to obtain
add: 4 new stations, all existing stations are unlockable
add trade bottle subtype to deal with the large number of bottles being added
del: cybernetics station (broken)
/:cl:

### CODE REVIEW

**Comment by garrulousTeamotei** [comment]
Am I reading the code right, or does the Guild gain access to NT cells before NT does?

**Comment by Pink-Chink** [comment]
Does NT not get cells at roundstart?

**Comment by garrulousTeamotei** [comment]
> Does NT not get cells at roundstart?
Both versions of the NT cells are eye rewards. 
They start with 3 disks, the NT basic melee, and a biofabrics and bioproducts
What's in those disks are in here https://github.com/discordia-space/CEV-Eris/blob/master/code/game/objects/items/weapons/design_disks/NeoTheology.dm
Everything Else NT are either eye rewards or just NT disks found in maintenance, non-exclusive.

**Comment by Pink-Chink** [comment]
Huh, I'll just bump the normal cells to the next tier up and cut the upgraded cells, then.

**Comment by garrulousTeamotei** [comment]
A thought, NT advanced seems underwhelming compared to NT basic. While animals can see some use for the bar, seeds are useless as can be for sale, where those who use them start with more than they need, and those who don't have the vendor in the public garden with still more seeds than needed, with the secret menu seeds available with some basic vendor hacking. Adding some of the randomly generated seeds available to Moe Xenobotany/Guild seed crate could spice up the secret menu. Still, seems lackluster for an unlockable station.
Might be better to bring all the weapons to the advanced NT station and rebrand it as an "inquisitorial ship" or along those lines, and bring the items from the advanced NT to the basic, as a supply ship.

**Comment by Pink-Chink** [comment]
Random seeds are available in the adv MoeSci trade station. Also, I could probably have it sell the actual seed vendor instead of individual seeds.

Since I don't want to allow the NT trade stations to overtake the actual department, I'm hesitant about giving the early stations department-specific stuff without hiking the price. With NT, there are quite a few items that seem weird to give Guild access to. Would the basic melee and grenades be too much?

**Comment by garrulousTeamotei** [comment]
>Would the basic melee and grenades be too much?

Basic melee seems fine for the basic NT non-secret. If I'm not mistaken, the shortsword is outclassed by even the junkblade, NT daggers are an easy maintenance find, sheaths are worthless, NT would love to see sanctifier tool mods distributed by any means. The buckler is in an odd place, small shields being useful and a NT/IH exclusive (is there a junk equivalent?), might be good for a basic NT secret menu low supply item. 
Selling any item by guild obtainable by the Eye will probably ruffle some feathers, due to how much a pain in the ass the Eye RNG is. grenades especially due to their again usefulness and NT/IH exclusivity. they'd probably have to be a advanced NT secret menu item, to force it to a late game item, with medium stock of flash/smoke and low supply of frag. Selling high explosive sounds like a bad idea.

**Comment by Pink-Chink** [comment]
I'm gonna toe the line and put NT shields and the non-frag, non-HE grenades in the uncommon secret inv. IH riot shields were already available via supply crates and original Hellcat.

**Comment by Humonitarian** [comment]
Do note there are two types of NT shields. The smaller buckler one is available roundstart, the other is considered late game gear.

**Comment by ValoTheValo** [comment]
more trade stations are always a good thing. are the prices at all being adjusted because stations overcharge currently?

**Comment by Pink-Chink** [comment]
> Do note there are two types of NT shields. The smaller buckler one is available roundstart, the other is considered late game gear.

Oops. Fixed the NT gear.

> more trade stations and overmap content are always a good thing. are the prices at all being adjusted because stations overcharge currently?

At a glance, things should be getting slightly cheaper across the board. I dropped the markup tiers from 1.2, 2.4, 4.8, and 9.6 to 1.2, 2.4, 3.6, and 4.8. If that's still too much, it's an easy change.

**Comment by garrulousTeamotei** [comment]
> Random seeds are available in the adv MoeSci trade station

Perhaps move the current secret seeds down to the rest of them, and have them sell mutant seeds, those seeds you can get when using unstable mutagen (icechili, glowberries, bluespacetomato, etc.).

**Comment by Pink-Chink** [comment]
Oh, I forgot about those. Would those be considered NT or Moe? Since the seed vending machine is in there now, I won't bother with anything already in there.

**Comment by garrulousTeamotei** [comment]
I'd say they'd be neutral. It's easiest to make them with unstable mutagen, so, moebius that way. But, they're still able to be made with radium, which can be generated by a plant in the seed vendor.

**Comment by Pink-Chink** [comment]
I'll put em in basic NT secret inv since the markup is lower.

File: code/modules/trade/datums/trade_stations_presets/1-common/zarya.dm [hunk 1]
```diff
@@ -65,11 +66,8 @@
...
-		/obj/item/tool/crowbar/onestar = offer_data("onestar crowbar", 1000, 3),
-		/obj/item/tool/pickaxe/onestar = offer_data("onestar pickaxe", 1000, 3),
-		/obj/item/tool/pickaxe/jackhammer/onestar = offer_data("onestar jackhammer", 1000, 3),
-		/obj/item/tool/screwdriver/combi_driver/onestar = offer_data("onestar combi driver", 1000, 3),
-		/obj/item/tool/weldingtool/onestar  = offer_data("onestar welding tool", 1000, 3),
-		/obj/item/tool_upgrade/augment/repair_nano = offer_data("repair nano", 5000, 1)
+		/obj/item/tool_upgrade = offer_data("tool upgrade", 200, 10),									// base price: 200
+		/obj/item/oddity/common/blueprint = offer_data("strange blueprint", 800, 1),
+		/obj/item/oddity/common/old_radio = offer_data("old radio", 800, 1),
+		/obj/item/organ/external/robotic/serbian = offer_data("serbian external prosthetic", 600, 8)	// base price: 600; roundstart item, but you'd be giving up an arm and a leg for cash
```
Comment (TheShown911) [inline code/modules/trade/datums/trade_stations_presets/1-common/zarya.dm:72]: Does this work, can you instance limbs like this?

**Review by TheShown911 — APPROVED** [review]
There may be some balance concerns, but this PR is way too cool.

File: code/modules/trade/datums/trade_stations_presets/1-common/zarya.dm (hunk 1 above)
Comment (Pink-Chink) [reply code/modules/trade/datums/trade_stations_presets/1-common/zarya.dm:72]: Yeah. At the moment, subtypes of an item count when evaluating offers and direct sells.

**Review by Firefox13 — APPROVED** [review]
very based

**Pull request merged**